consensus/src/experimental/execution_phase.rs

```diff
@@ -1,121 +1,118 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{
-    experimental::{commit_phase::CommitChannelType, errors::Error},
-    state_replication::{StateComputer, StateComputerCommitCallBackType},
-};
-use channel::{Receiver, Sender};
+use crate::state_replication::StateComputer;
+use anyhow::Result;
 use consensus_types::{block::Block, executed_block::ExecutedBlock};
-use diem_types::ledger_info::LedgerInfoWithSignatures;
+use diem_crypto::HashValue;
 use executor_types::Error as ExecutionError;
-use futures::{channel::oneshot, select, FutureExt, SinkExt, StreamExt};
-use std::sync::Arc;
+use futures::{
+    channel::mpsc::{UnboundedReceiver, UnboundedSender},
+    select, SinkExt, StreamExt,
+};
+use std::{
+    fmt::{Debug, Display, Formatter},
+    sync::Arc,
+};
+use thiserror::Error;
 
 /// [ This class is used when consensus.decoupled = true ]
 /// ExecutionPhase is a singleton that receives ordered blocks from
-/// the ordering state computer and execute them. After the execution is done,
-/// ExecutionPhase sends the ordered blocks to the commit phase.
+/// the buffer manager and execute them. After the execution is done,
+/// ExecutionPhase sends the ordered blocks back to the buffer manager.
 ///
 
-pub type ResetAck = ();
-pub fn reset_ack_new() -> ResetAck {}
-
-pub struct ExecutionChannelType(
-    pub Vec<Block>,
-    pub LedgerInfoWithSignatures,
-    pub StateComputerCommitCallBackType,
-);
+pub struct ExecutionIncomingType {
+    pub blocks: Vec<Block>,
+}
 
-impl std::fmt::Debug for ExecutionChannelType {
+impl Debug for ExecutionIncomingType {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         write!(f, "{}", self)
     }
 }
 
-impl std::fmt::Display for ExecutionChannelType {
+impl Display for ExecutionIncomingType {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        write!(f, "ExecutionChannelType({:?}, {})", self.0, self.1)
+        write!(f, "ExecutionChannelType({:?})", self.blocks)
     }
 }
 
+pub struct ExecutionOutgoingType {
```

Comment:
similarly ExecutionPhaseResponse, this can be a type alias too, type ExecutionPhaseResponse = Result<ExecutionBlock, Error>

Reply:
I see. But probably later we need to add more fields/methods to it. Let's keep it as a struct.